conan/tools/google/layout.py

```diff
@@ -1,11 +1,13 @@
+import os
 
 
-def bazel_layout(conanfile, generator=None, src_folder="."):
-    """The layout for bazel is very limited, it builds in the root folder even specifying
-       "bazel --output_base=xxx" in the other hand I don't know how to inject a custom path so
-       the build can load the dependencies.bzl from the BazelDeps"""
-    conanfile.folders.build = ""
-    conanfile.folders.generators = ""
-    # used in test package for example, to know where the binaries are (editables not supported yet)
-    conanfile.cpp.build.bindirs = ["bazel-bin"]
-    conanfile.cpp.build.libdirs = ["bazel-bin"]
+def bazel_layout(conanfile, src_folder="."):
+    """Bazel layout is so limited. It does not allow to create its special symlinks in other
+    folder. See more information in https://bazel.build/remote/output-directories"""
+    subproject = conanfile.folders.subproject
+    conanfile.folders.source = src_folder if not subproject else os.path.join(subproject, src_folder)
+    conanfile.folders.build = "."  # Bazel always build the whole project in the root folder
+    conanfile.folders.generators = "bazel-conan-tools"  # one
```

Comment:
Minor thing, but what about changing the name of the folder to `conan` or `generators` like in other generators layouts?

```suggestion
    conanfile.folders.generators = "conan"  # one
```